base/sysinfo.jl

```diff
@@ -567,4 +568,25 @@ function which(program_name::String)
 end
 which(program_name::AbstractString) = which(String(program_name))
 
+"""
+    Sys.username() -> String
+
+Return the username for the current user. If the username cannot be determined
+or is empty, this function throws an error.
+
+To retrieve a username that is overridable via an environment variable,
+e.g., `USER`, consider using
+```julia
+user = get(Sys.username, ENV, "USER")
+```
+
+!!! compat "Julia 1.11"
+    This function requires at least Julia 1.11.
+"""
+function username()
+    pw = Libc.getpw()
+    isempty(pw.username) && throw(ErrorException("empty username"))
```

Comment:
```suggestion
    isempty(pw.username) && throw(Base.IOError("empty username", Base.UV_ENOENT))
```
So, this is preferred, @vtjnash?

Reply:
Maybe like this:
```suggestion
    isempty(pw.username) && Base.uv_error("username", Base.UV_ENOENT)
```
to match this behavior:
```
julia> Libc.getpwuid(typemax(UInt))
ERROR: IOError: getpwuid: no such file or directory (ENOENT)
```